### PR TITLE
adapt status/sync to all container types (async only)

### PR DESCRIPTION
**Pull request by pkasprzyk**

### CODE REVIEW

File: e2e_tests/standard/test_sync.py [hunk 1]
```diff
@@ -17,134 +17,103 @@
...
 
 runner = CliRunner()
 
 
 class TestSync(BaseE2ETest):
-    SYNCHRONIZED_SYSID_RE = r"\w+/[\w-]+/([\w-]+)"
-
-    def test_sync_run(self, environment):
-        custom_run_id = "-".join((fake.word() for _ in range(3)))
+    SYNCHRONIZED_SYSID_RE = r"[\w-]+/[\w-]+/([\w-]+)"
```
Comment (shnela) [inline e2e_tests/standard/test_sync.py:38]: What was wrong with this RE?

File: neptune/new/sync/__init__.py [hunk 2]
```diff
@@ -115,42 +122,56 @@ def status(path: Path) -> None:
...
     # to project "workspace/project"
-    neptune sync --project workspace/project --run offline/a1561719-b425-4000-a65a-b5efb044d6bb
+    neptune sync --project workspace/project --object offline/a1561719-b425-4000-a65a-b5efb044d6bb
     """
 
     sync_runner = SyncRunner(backend=HostedNeptuneBackend(Credentials.from_token()))
 
     if runs_names:
-        sync_runner.sync_selected_runs(path, project_name, runs_names)
+        print("DAUPDUAPIDPOSDDSA")
```
Comment (shnela) [inline neptune/new/sync/__init__.py:166]: Please, don't :p

File: neptune/new/sync/__init__.py [hunk 3]
```diff
@@ -115,42 +122,56 @@ def status(path: Path) -> None:
...
+        click.echo(
+            "WARNING: --run parameter is deprecated and will be removed in the future, please start using --object"
+        )
+        # prefer object_names, obviously
+        if not object_names:
```
Comment (shnela) [inline neptune/new/sync/__init__.py:171]: I think there was a comment from Mycek, that we finally decided to accept both `--run` and `--object` parameters mixed, but here you accept only one of them.

File: neptune/new/sync/status.py [hunk 4]
```diff
@@ -54,76 +54,86 @@ def partition_containers(
...
+            [
+                exp
+                for exp in synced_containers + unsynced_containers
+                if not exp or exp.trashed
+            ]
         )
-        synced_runs = [exp for exp in synced_runs if exp and not exp.trashed]
-        unsynced_runs = [exp for exp in unsynced_runs if exp and not exp.trashed]
+        synced_containers = [
+            exp for exp in synced_containers if exp and not exp.trashed
```
Comment (shnela) [inline neptune/new/sync/status.py:80]: Let's get rid of `exp` since it's not client-domain name.

File: neptune/new/sync/utils.py [hunk 5]
```diff
@@ -104,11 +107,11 @@ def get_project(
         return None
 
 
-def get_qualified_name(run: ApiExperiment) -> str:
-    return "{}/{}/{}".format(run.workspace, run.project_name, run.sys_id)
+def get_qualified_name(run: ApiExperiment) -> QualifiedName:
```
Comment (shnela) [inline neptune/new/sync/utils.py:110]: Nice typing

**Review by shnela — CHANGES_REQUESTED** [review]

File: e2e_tests/standard/test_sync.py (hunk 1 above)
Comment (pkasprzyk) [reply e2e_tests/standard/test_sync.py:38]: Wasn't matching workspace `e2e-tests`, for instance (which I was using ;)). This one is simply more robust.

File: e2e_tests/standard/test_sync.py [hunk 6]
```diff
@@ -17,134 +17,103 @@
...
+                container_id = container._id
+                container_sys_id = container._sys_id
+
+            # manually add operations to queue
+            queue_dir = list(
+                Path(f"./.neptune/async/{container_type}__{container_id}/").glob(
+                    "exec-*"
                 )
+            )[0]
+            with open(
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply e2e_tests/standard/test_sync.py:64]: I'm using container in `with` context manager each time, so it does `.stop()` on exit

File: e2e_tests/standard/test_sync.py [hunk 7]
```diff
@@ -167,9 +136,10 @@ def test_offline_sync(self, environment):
             assert result.exit_code == 0
 
             # offline mode doesn't support custom_run_id, we'll have to parse sync output to determine short_id
+            print(result.stdout)
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply e2e_tests/standard/test_sync.py:139]: yep, sorry

File: neptune/new/sync/__init__.py (hunk 2 above)
Comment (pkasprzyk) [reply neptune/new/sync/__init__.py:166]: sorry :D

File: neptune/new/sync/__init__.py (hunk 3 above)
Comment (pkasprzyk) [reply neptune/new/sync/__init__.py:171]: Oh god, hate that. But ok, I guess

File: e2e_tests/standard/test_sync.py [hunk 8]
```diff
@@ -17,134 +17,103 @@
...
-        assert exp3[key].fetch() == updated_value
-        assert exp3["copy/" + key].fetch() == updated_value
+            with reinitialize_container(
+                container_sys_id, container_type, project=environment.project
+            ) as container:
+                # and we should get the updated value from server
+                assert container[key].fetch() == updated_value
+                assert container["copy/" + key].fetch() == updated_value
 
     def test_offline_sync(self, environment):
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply e2e_tests/standard/test_sync.py:118]: right, added

File: neptune/new/sync/__init__.py [hunk 9]
```diff
@@ -163,14 +163,14 @@ def sync(
...
         click.echo(
             "WARNING: --run parameter is deprecated and will be removed in the future, please start using --object"
         )
         # prefer object_names, obviously
-        if not object_names:
-            object_names = runs_names
+        object_names = set(object_names)
+        object_names.update(runs_names)
 
+    print(object_names)
```
Comment (shnela) [inline neptune/new/sync/__init__.py:173]: print again :p

File: e2e_tests/standard/test_sync.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (shnela) [reply e2e_tests/standard/test_sync.py:64]: Right

**Review by shnela — APPROVED** [review]

**Pull request merged**